src/PowerShellEditorServices/Services/Extension/ExtensionService.cs

```diff
@@ -91,16 +93,25 @@ internal sealed class ExtensionService
         /// </summary>
         /// <param name="editorOperations">An IEditorOperations implementation.</param>
         /// <returns>A Task that can be awaited for completion.</returns>
-        internal async Task InitializeAsync()
+        internal Task InitializeAsync()
         {
+            if (Interlocked.Exchange(ref _initialized, 1) != 0)
```

Comment:
We use this pattern in a few places and it might be worth wrapping it in a struct for reuse